docs/rfcs/0561-list-metadata-reuse.md

```diff
@@ -0,0 +1,131 @@
+- Proposal Name: `list_metadata_reuse`
+- Start Date: 2022-08-23
+- RFC PR: [datafuselabs/opendal#561](https://github.com/datafuselabs/opendal/pull/561)
+- Tracking Issue: None
+
+# Summary
+
+Extend `DirEntry` with more `metadata` fields.
```

Comment:
This `summary` is incorrect. We intend to reuse the metadata that returned during `list`. Maybe we could change this into the following:

Reuse metadata returned during list by extending `DirEntry` with more metadata fields.